describe.ts

```diff
@@ -195,6 +195,98 @@ function it(
+  only?: boolean;
+  /**
+   * Check that the number of async completed ops after the test is the same as
+   * the number of dispatched ops after the test. Defaults to true.
+   */
+  sanitizeOps?: boolean;
+  /**
+   * Ensure the test case does not "leak" resources - ie. the resource table after the test
+   * has exectly the same contents as before the test. Defaults to true.
+   */
+  sanitizeResources?: boolean;
+  /**
+   * The cases to execute the test for
+   */
+  cases: EachCaseType<T>[];
+}
+/**
+ * Generate a set of identical tests with different parameters (e.g. to test
+ * different inputs to your code under test without copy&pasting the test code).
+ *
+ * @param name The base name of the test case to register. This will be used as
+ *  the prefix for the actual test case name, which is suffixed with either the
+ *  current parameter set or the parameter set name
+ * @param cases The different test case parameter sets. Each parameter set is an
+ *  array giving the parameters for one call to it().
+ *
+ *  Each entry in the array can be one of two things:
+ *  1. an array of parameters for this one particular test case
+ *  2. an object of the form { name: string, params: unknown[] }
+ *
+ * In the first case, tests case names will be generated by suffixing the base
+ * name with the parameter array. This is suitable for test cases, where your
+ * parameters are not too large when printed. In the second case, test cases
+ * will be suffixed with the name field of the given object, whereas the
+ * params field contains the actual test case parameters. This is for cases
+ * where using your parameters as test names would lead to very long names and
+ * make reading the test logs very hard.
+ * @param fn The test function. This function must have parameters corresponding
+ *  to the entries in one parameter set in the cases array.
+ */
+function each<T extends unknown[]>(
+  name: string,
+  cases: EachCaseType<T>[],
+  fn: (...params: T) => void | Promise<void>,
+): void;
+function each<T extends unknown[]>(options: EachDefinition<T>): void;
+function each<T extends unknown[]>(
+  a: string | EachDefinition<T>,
+  cases?: EachCaseType<T>[],
+  fn?: (...params: T) => void | Promise<void>,
+): void {
+  if (currentSuite) lockHooks();
+  let myOptions: EachDefinition<T>;
+  if (typeof a === "string") {
+    myOptions = { name: a, fn: fn!, cases: cases! };
+  } else {
+    myOptions = a;
+  }
+
+  myOptions.cases.forEach((c) => {
+    const testOptions: TestDefinition<void> = {
+      name: Array.isArray(c)
+        ? `${myOptions.name}: ${c}`
+        : `${myOptions.name}: ${c.name}`,
+      fn: Array.isArray(c)
+        ? () => myOptions.fn(...c)
+        : () => myOptions.fn(...c.params),
+      ignore: myOptions.ignore,
+      only: myOptions.only,
+      sanitizeOps: myOptions.sanitizeOps,
+      sanitizeResources: myOptions.sanitizeResources,
+    };
+    if (currentSuite) testOptions.suite = currentSuite;
+    test(testOptions);
```

Comment:
This could be simplified by just calling `it`. Since `it` will lockHooks, you don't need the call to lockHooks at the start of `each`.

```ts
myOptions.cases.forEach((c) => it({
  name: Array.isArray(c)
    ? `${myOptions.name}: ${c}`
    : `${myOptions.name}: ${c.name}`,
  fn: Array.isArray(c)
    ? () => myOptions.fn(...c)
    : () => myOptions.fn(...c.params),
  ignore: myOptions.ignore,
  only: myOptions.only,
  sanitizeOps: myOptions.sanitizeOps,
  sanitizeResources: myOptions.sanitizeResources,
}))
```

Reply:
Thanks for the feedback. I've changed the call accordingly.